### PR TITLE
Adds promoted/flagged reasons to the reportback view

### LINKED ISSUE #4753: Add new OP/column to reportback log
In the change log on the admin view of a reportback, it should include the new OP rows as well as the new 'flagged/promoted' OPs

screen shot 2015-07-28 at 9 54 58 am

### PR DESCRIPTION
**Pull request by itsjoekent**
Fixes #4753 

Adds new columns for promoted and flagged reasons 

@angaither

### CODE REVIEW

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.views_default.inc [hunk 1]
```diff
@@ -130,18 +130,21 @@ function dosomething_reportback_views_default_views() {
   $handler->display->display_options['fields']['status']['id'] = 'status';
   $handler->display->display_options['fields']['status']['table'] = 'dosomething_reportback_file';
   $handler->display->display_options['fields']['status']['field'] = 'status';
-  $handler->display->display_options['fields']['status']['label'] = 'Item Status';
-  /* Field: Reportback: Flagged status */
-  $handler->display->display_options['fields']['flagged']['id'] = 'flagged';
-  $handler->display->display_options['fields']['flagged']['table'] = 'dosomething_reportback';
-  $handler->display->display_options['fields']['flagged']['field'] = 'flagged';
-  $handler->display->display_options['fields']['flagged']['relationship'] = 'rbid';
-  $handler->display->display_options['fields']['flagged']['label'] = 'Reportback flagged status';
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.views_default.inc:139]: I don't think this stuff should be removed, maybe your feature is out of date. Try reverting your feature before adding your changes.

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.views_default.inc:139]: gotcha -- ill just make it again cause its quick but not sure how to remove the feature first?

Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.views_default.inc:139]: you don't need to remove the feature, you have to revert it. you can use drush `drush fr dosomething_reportback -y --force`

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.info [hunk 2]
```diff
@@ -21,4 +21,4 @@ features[user_permission][] = view any reportback
 features[user_permission][] = view own reportback
 features[views_view][] = reportback_files
 files[] = dosomething_reportback.test
-mtime = 1438264043
+mtime = 1439822699
```
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.info]: Should this file still be in the commit history?

Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.info]: no, not if you didn't change anything in the view

File: lib/modules/dosomething/dosomething_reportback/includes/ReportbackController.php [hunk 3]
```diff
@@ -57,11 +57,11 @@ public function buildContent($entity, $view_mode = 'full', $langcode = NULL, $co
 
     // Output Reportback Log.
     $rows = array();
-    $header = array('Submitted', 'Op', 'Uid', 'Files', 'IP', 'Quantity', 'Why Participated');
+    $header = array('Submitted', 'Op', 'Uid', 'Files', 'IP', 'Quantity', 'Why Participated', 'Reasons');
     foreach ($entity->getReportbackLog() as $delta => $record) {
       $submitted = format_date($record->timestamp, 'short');
       $why = check_plain($record->why_participated);
-      $rows[] = array($submitted, $record->op, $record->uid, $record->files, $record->remote_addr, $record->quantity, $why);
+      $rows[] = array($submitted, $record->op, $record->uid, $record->files, $record->remote_addr, $record->quantity, $why, $record->reason);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/includes/ReportbackController.php:64]: does this throw errors in the dblog if it isn't set?

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_reportback/includes/ReportbackController.php:64]: Nope, I don't see any errors caused by this in the log

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.info (hunk 2 above)
Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.info]: can you remove this file from the changes you're pushing?

**Comment by angaither** [comment]
once that's done :+1:

**Comment by itsjoekent** [comment]
removed info file from history

**Pull request merged**